extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php

```diff
@@ -56,11 +56,48 @@ public static function onBeforeInitialize( Title $title, $unused, OutputPage $ou
 				!self::httpsEnabledTitle( $title )
 			) {
 				$output->redirect( wfHttpsToHttp( $requestURL ) );
+				$output->enableClientCache( false );
 			}
 		}
 		return true;
 	}
 
+	/**
+	 * Handle downgrading anonymous requests for our sitemaps.
+	 *
+	 * @param  string     $subpage Specific sitemap being requested.
+	 * @param  WebRequest $request
+	 * @param  User       $user
+	 * @return boolean
+	 */
+	public static function onSitemapPageBeforeOutput( string $subpage, WebRequest $request, User $user ): bool {
+		global $wgScriptPath;
+		if ( WebRequest::detectProtocol() === 'https' &&
+			!self::httpsAllowed( $user, $request->getFullRequestURL() )
+		) {
+			self::downgradeRedirectForPath( "$wgScriptPath/$subpage", $request );
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Handle downgrading anonymous requests for robots.txt.
+	 *
+	 * @param  WebRequest $request
+	 * @param  User       $user
+	 * @return boolean
+	 */
+	public static function onWikiaRobotsBeforeOutput( WebRequest $request, User $user ): bool {
```

Comment:
This looks very similar to `onSitemapPageBeforeOutput`. Maybe we could create a new method that takes the `$user`, `$request` and `redirectUrl` params. And ideally if we manage to solve the issues of undoing VCL redirects, the `redirectUrl` won't be needed anymore, as we will be able to use `getFullRequestURL`